crates/wasmprinter/src/lib.rs

```diff
@@ -1385,32 +1396,41 @@ impl Printer<'_, '_> {
         // with the closing paren printed for the func.
         if self.nesting != nesting_start {
             self.nesting = nesting_start;
-            self.newline(body.original_position())?;
+            self.newline(reader.original_position())?;
         }
 
         Ok(())
     }
 
     fn print_operators<'a, O: OpPrinter>(
         body: &mut BinaryReader<'a>,
+        data_index_allowed: bool,
```

Comment:
re my question above about this as well - this is an example of where it's a bit cumbersome to manage this and I'd prefer to remove this if possible, but if it's required by tests I mostly just want to understand in more detail what's requiring it.

Reply:
Understood -- discussed at https://github.com/bytecodealliance/wasm-tools/pull/2134#discussion_r2038328446